src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,198 @@
+            };
+
+            entryPoint.Tests = Tests;
+
+            await entryPoint.RunAsync();
+
+            if (File.Exists(entryPoint.TestsResultsFinalPath))
+            {
+                bundle.PutString(ResultsPathArgumentName, entryPoint.TestsResultsFinalPath);
+            }
+            
+            if (bundle.GetLong(ExitCodeArgumentName, -1) == -1)
+            {
+                bundle.PutLong(ExitCodeArgumentName, 1);
+            }
+
+            Finish(Result.Ok, bundle);
+        }
+
+        class TestsEntryPoint : AndroidApplicationEntryPoint
+        {
+            readonly string resultsPath;
+            readonly string includeMethod;
+            readonly string excludeCategories;
+            readonly string excludeMethod;
+            readonly string includeClass;
+            readonly string excludeClass;
+
+            public TestsEntryPoint(Dictionary<string, string> arguments)
+            {
+                var root = Application.Context.GetExternalFilesDir(null)?.AbsolutePath;
+
+                var docsDir = Path.Combine(root, "Documents");
+
+                if (!Directory.Exists(docsDir))
+                {
+                    Directory.CreateDirectory(docsDir);
+                }
+
+                resultsPath = Path.Combine(docsDir, arguments[ResultsFileArgumentName]);
+                includeMethod = arguments[IncludeMethodArgumentName];
+                excludeCategories = arguments[ExcludeCategoriesArgumentName];
+                excludeMethod = arguments[ExcludeMethodArgumentName];
+                includeClass = arguments[IncludeClassArgumentName];
+                excludeClass = arguments[ExcludeClassArgumentName];
+            }
+
+            protected override bool LogExcludedTests => true;
+
+            public override TextWriter Logger => null;
+
+            public override string TestsResultsFinalPath => resultsPath;
+
+            protected override int? MaxParallelThreads => System.Environment.ProcessorCount;
+
+            protected override IDevice Device => null;
+
+            public IEnumerable<string> Tests { get; internal set; }
+
+            protected override IEnumerable<TestAssemblyInfo> GetTestAssemblies()
+            {
+                //yield return new TestAssemblyInfo(Assembly.GetExecutingAssembly(), Assembly.GetExecutingAssembly().Location);
+                //yield return new TestAssemblyInfo(typeof(Battery_Tests).Assembly, typeof(Battery_Tests).Assembly.Location);
+                foreach (string file in Tests)
+                {
+                    yield return new TestAssemblyInfo(Assembly.LoadFrom(file), file);
+                }
+            }
+            protected override void TerminateWithSuccess()
+            {
+            }
+
+            protected override TestRunner GetTestRunner(LogWriter logWriter)
+            {
+                var testRunner = base.GetTestRunner(logWriter);
```

Comment:
Please compare with the `iOSApplicationEntryPoint`. When we were ignoring some methods, we would also flip some flag saying "do not execute all tests". Not sure we should do the same here

Reply:
I added handling with `runAllTests`